### DISCUSSION TITLE
Relationship queries ignore global scopes during force delete

### DISCUSSION THREAD

**Discussion by fadarrizz** [laravel/framework#53402] [General] [answered]
I found out when calling `forceDelete` on a relationship query, it deletes all related models even if a global scope is active.
This was a huge surprise to me and resulted in production data being permanently deleted.

For example, let's say we have a model called Foo and a model called Bar, with Foo having many Bar's. Bar is soft deletable.

We have one Foo row and two Bar rows in the database, both Bar rows related to Foo. Bar 1 is soft deleted (deleted_at not null), Bar 2 is not soft deleted (deleted_at null).

When we run `$foo->bar()->toSql()`, we'll see the following sql: 
```sql
select * from `foo` where `bar`.`foo_id` = ? and `bar`.`foo_id` is not null and `bar`.`deleted_at` is null
```
The query shows the soft deletes scope being active.  

When we then call `$foo->bar()->forceDelete()`, we see that both Bar rows are deleted from the database.

Why is this the case, and more importantly, why can't I find this in the documentation anywhere? Did I miss something?

**Comment by iazaran** [comment]
`forceDelete` on a relationship query bypasses any global scopes, including the soft delete scope.
To prevent this from happening, you can manually filter the records:
```php
$barsToDelete = $foo->bar()->whereNotNull('deleted_at');
$barsToDelete->forceDelete();
```

**Reply by fadarrizz** [reply]
Thanks for replying @iazaran.

How do you know that it works this way? Did you find out by experience or did you read it somewhere in the documentation? I'm really curious, because I couldn't read anything about this.

I don't understand why `forceDelete` should bypass any global scope, besides `SoftDeletes`. For such important functionality, these things should not come as a surprise and should be known to the user or bad things can happen, as it did in my case.

**Reply by iazaran** [reply, marked as answer]
Unfortunately it is not in docs and I had this experience myself too.
In most cases in different languages, I see `force` and they don't care about the details, so because of this they warn to use this word. Like `forceFill` that ignore all conditions in the model.
